packages/compartment-mapper/src/import-hook.js

```diff
@@ -151,6 +155,35 @@ export const makeImportHookMaker = (
       const { read } = unpackReadPowers(readPowers);
 
       for (const candidateSpecifier of candidates) {
+        const candidateModuleDescriptor = moduleDescriptors[candidateSpecifier];
+        if (candidateModuleDescriptor !== undefined) {
+          const { compartment: candidateCompartmentName = packageLocation } =
+            candidateModuleDescriptor;
+          const candidateCompartment = compartments[candidateCompartmentName];
+          if (candidateCompartment === undefined) {
+            throw new Error(
+              `compartment missing for candidate ${candidateSpecifier} in ${candidateCompartmentName}`,
+            );
+          }
+          // modify compartmentMap to include this redirect
+          const candidateCompartmentDescriptor =
+            compartmentDescriptors[candidateCompartmentName];
+          if (candidateCompartmentDescriptor === undefined) {
+            throw new Error(
+              `compartmentDescriptor missing for candidate ${candidateSpecifier} in ${candidateCompartmentName}`,
+            );
+          }
+          candidateCompartmentDescriptor.modules[moduleSpecifier] =
+            candidateModuleDescriptor;
```

Comment:
```suggestion
          // Leave a note in the compartment map for archives generated from this load.
          candidateCompartmentDescriptor.modules[moduleSpecifier] =
            candidateModuleDescriptor;
```

Reply:
this comment is already present, but its maybe a little too far up
```js
// modify compartmentMap to include this redirect
```